test/protocol.spec.ts

```diff
@@ -27,11 +27,12 @@ describe('protocol-contract', async () => {
     });
 
     r = await d.protocol.createDeploymentSubset(DEPLOYMENT_SUBSET_CANARY, 2);
+    currTime = await getTopBlockTimestamp(d);
     expect(r).to.have.a.protocolChangedEvent({
       deploymentSubset: DEPLOYMENT_SUBSET_CANARY,
       currentVersion: bn(2),
       nextVersion: bn(2),
-      fromTimestamp: bn(r.blockNumber)
+      fromTimestamp: bn(currTime) // assuming ganache will not mine since last tx
```

Comment:
You can take the exact timestamp from the transaciton receipt (`r`). I think there should be somewhere a function named `txTimestamp`.

Reply:
Couldn't find it. the only way I saw proposed in stack overflow is to get the block number, and then to ask the timestamp of the block which is basically what I'm doing here.

If I missed something let me know and I'll do another PR...

Reply:
See `txTimestamp` in eth.ts, it does exactly that. The difference is that it take the timestamp of the block number of the transaction, and you take the timestamp from the latest block number (which may have changed).